src/daemon/devnode.c

```diff
@@ -101,8 +130,9 @@ int _mknotifynode(usbdevice* kb, int notify){
         remove(outpath);
         return -1;
     }
-    if(gid >= 0)
-        fchown(kb->outfifo[notify] - 1, 0, gid);
+    if(gid >= 0) {
+        return check_fchown(kb->outfifo[notify] - 1, 0, gid);
```

Comment:
Is there ever a case where `gid` isn't >= 0? If not, this should just be `return check_fchown(...)`.